### DISCUSSION TITLE
PrimeFaces 12 → 13 upgrade issue

### DISCUSSION THREAD

**Discussion by sivarajka** [primefaces/community#3018] [PrimeFaces] [answered]
Hi All,

We have been using primefaces version 12 so far in our project. Due to some vulnerability issue with json package we had to upgrade it to 13.0.10. Once we upgraded the functionalities are working as expected. But we seeing one error message in the logs.

Nov 13, 2024 11:50:20 AM org.primefaces.component.datatable.DataTable convertIntoObjectValueType
SEVERE: com.fis.dnweb.accesssecurity.bean.GenericModel.<init>(java.util.List)

Could any body help on this? How to resolve the error message.

```java
public class GenericModel<T> extends ListDataModel<T> implements SelectableDataModel<T>, Serializable {
	private static final long serialVersionUID = 1L;

	@Override
	public String getRowKey(T object) {
		return object.toString().trim();
	}

	@SuppressWarnings({ "rawtypes", "unchecked" })
	@Override
	public T getRowData(String rowKey) {
		for (Object o : (List) getWrappedData()) {
			if (o.toString().trim().equals(rowKey.trim())) {
				return (T) o;
			}
		}
		return null;
	}

}
```

Thanks
Sivaraj

**Comment by tandraschko** [comment]
did you check the migration guide?
anyway, i can just suggest to use 14.0.x versions. each version is more stable. its really nosense to upgrade to 13.x when 14.x is already out.

**Reply by melloware** [reply]
Also your error has no information in it?

**Reply by sivarajka** [reply]
Hi - We are still running on Java1.8. PF 14 doesn't support Java 8

**Reply by sivarajka** [reply]
This is the only message we are seeing in the console. 
Nov 13, 2024 11:50:20 AM org.primefaces.component.datatable.DataTable convertIntoObjectValueType
SEVERE: com.fis.dnweb.accesssecurity.bean.GenericModel.(java.util.List)
Is there any way to hide these messages?

**Reply by Rapster** [reply, marked as answer]
SEVERE: com.fis.dnweb.accesssecurity.bean.GenericModel.(java.util.List) => It says missing constructor taking a List as parameter
